### PR TITLE
Author System

### PR DESCRIPTION
**Pull request by IMB11**
Allows users to be credited for the pages they've written/contributed to. Should encourage contributions.

image

Simply add a github username to the `authors` array in the markdown frontmatter.

I tried out a git system, but it was extremely messy, so manual it is.

### CODE REVIEW

**Review by Superkat32 — DISMISSED** [review]

File: .vitepress/theme/components/PageAuthorComponent.vue [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  if (pageData.frontmatter.value["authors"]) {
+    authors.value = pageData.frontmatter.value["authors"];
+  }
+}
+
+onContentUpdated(() => {
+  refreshData();
+});
+
+// router.onAfterRouteChanged = () => {
```
Comment (modmuss50) [inline .vitepress/theme/components/PageAuthorComponent.vue]: Remove commented out code.

File: .vitepress/theme/components/PageAuthorComponent.vue [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+function refreshData() {
+  authors.value = [];
+
+  if (pageData.frontmatter.value["authors"]) {
+    authors.value = pageData.frontmatter.value["authors"];
+  }
+}
+
+onContentUpdated(() => {
```
Comment (modmuss50) [inline .vitepress/theme/components/PageAuthorComponent.vue:30]: Is this not statically built? If not whats the reasoning why it has to be done on the client side like this?

Comment (IMB11) [reply .vitepress/theme/components/PageAuthorComponent.vue:30]: It is static. This is automatically prerendered during the build stage.

File: .vitepress/theme/components/PageAuthorComponent.vue (hunk 1 above)
Comment (IMB11) [reply .vitepress/theme/components/PageAuthorComponent.vue]: Done.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**